src/diffusers/quantizers/gguf/utils.py

```diff
@@ -0,0 +1,384 @@
+# Copyright 2024 The HuggingFace Team and City96. All rights reserved.
+# #
+# # Licensed under the Apache License, Version 2.0 (the "License");
+# # you may not use this file except in compliance with the License.
+# # You may obtain a copy of the License at
+# #
+# #     http://www.apache.org/licenses/LICENSE-2.0
+# #
+# # Unless required by applicable law or agreed to in writing, software
+# # distributed under the License is distributed on an "AS IS" BASIS,
+# # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# # See the License for the specific language governing permissions and
+# # limitations under the License.
+
+
+import gguf
+import torch
+import torch.nn as nn
+
+
+def _replace_with_gguf_linear(model, compute_dtype, state_dict, prefix=""):
+    def _should_convert_to_gguf(module, state_dict, prefix):
+        weight_key = prefix + "weight"
+        return weight_key in state_dict and isinstance(state_dict[weight_key], GGUFParameter)
+
+    has_children = list(model.children())
+    if not has_children:
+        return
+
+    for name, module in model.named_children():
+        module_prefix = prefix + name + "."
+        _replace_with_gguf_linear(module, compute_dtype, state_dict, module_prefix)
+
+        if isinstance(module, nn.Linear) and _should_convert_to_gguf(module, state_dict, module_prefix):
```

Comment:
Should we keep a placeholder to have `modules_to_not_convert`? Could be very important for video models, no?